puppetboard/app.py

```diff
@@ -142,15 +142,21 @@ def index():
             stats['noop'] += 1
         else:
             stats['unchanged'] += 1
+        reports = get_or_abort(puppetdb._query, 'reports',
+                               query='["=","certname","{0}"]'.format(node.name),
+                               limit=1)
+        if len(reports) > 0:
+            report_status[node.name] = reports[0]['status']
```

Comment:
The Reports endpoint in PuppetDB 3.0 and above support the latest_report? query field. Please modify your query to look something like this (Can remove the limit argument too):

'["and", ["=", "certname", "{0}"], ["=", "latest_report?", true]]'